### ISSUE TITLE
docker-machine ssh command with flags seem to not work anymore

### ISSUE THREAD

**Issue by nathanleclaire** [docker/machine#1039]
@sthulb @ehazlett 

Try this with an older version of `docker-machine` (probably 0.2.0 will suffice):

```
$ docker-machine ssh name -- df -h
Filesystem      Size  Used Avail Use% Mounted on
/dev/vda1        20G  2.9G   16G  16% /
none            4.0K     0  4.0K   0% /sys/fs/cgroup
udev            235M  4.0K  235M   1% /dev
tmpfs            50M  328K   49M   1% /run
none            5.0M     0  5.0M   0% /run/lock
none            246M     0  246M   0% /run/shm
none            100M     0  100M   0% /run/user
```

But with the recent SSH changes doing such commands with flags and flag parsing terminator seems to no longer work:

```console
$ ./docker-machine_darwin-amd64 ssh name -- df -h
bash: --: invalid option
Usage:  bash [GNU long option] [option] ...
        bash [GNU long option] [option] script-file ...
GNU long options:
        --debug
        --debugger
        --dump-po-strings
        --dump-strings
        --help
        --init-file
        --login
        --noediting
        --noprofile
        --norc
        --posix
        --rcfile
        --restricted
        --verbose
        --version
Shell options:
        -ilrsD or -c command or -O shopt_option         (invocation only)
        -abefhkmnptuvxBCHP or -o option
FATA[0000] Process exited with: 2. Reason was:  ()
```

**Comment by sthulb** [comment]
@nathanleclaire Thanks! I'll create a patch for @ehazlett to merge tomorrow

**Comment by nathanleclaire** [comment]
:+1:

**Comment by sthulb** [comment]
There seems to be a problem with codegansta/cli parsing things after `--`

**Comment by ehazlett** [comment]
This is fixed in #1046.  Thanks!
